resteasy-jaxrs/src/main/java/org/jboss/resteasy/plugins/delegates/ClientCookieHeaderDelegate.java

```diff
@@ -0,0 +1,29 @@
+package org.jboss.resteasy.plugins.delegates;
+
+import javax.ws.rs.core.NewCookie;
+
+import org.jboss.resteasy.resteasy_jaxrs.i18n.Messages;
+
+/**
+ * 
+ * @author <a href="mailto:ron.sigal@jboss.com">Ron Sigal</a>
+ * @date May 20, 2016
+ */
+public class ClientCookieHeaderDelegate extends NewCookieHeaderDelegate
+{
+   public String toString(Object value)
+   {
+      if (value == null)
+      {
+         throw new IllegalArgumentException(Messages.MESSAGES.paramNull());
+      }
+      NewCookie cookie = (NewCookie) value;
+      StringBuilder b = new StringBuilder();
+      b.append(cookie.getName()).append('=');
+      if (cookie.getValue() != null)
+      {
+         quote(b, cookie.getValue());
```

Comment:
This is serializing "version 0" cookies (which happens to match what the web uses, and has been spec'd in RFC 6265), but the JAX-RS spec relies on RFC 2965 (version 1 cookies, never actually deployed by anyone).
So while this works (except values aren't quoted, but their value range restricted), this breaks spec compliance.
I've filed a bug against the spec a while ago, left without any response.